programs/exchange_api/src/exchange_processor.rs

```diff
@@ -296,22 +296,22 @@ impl ExchangeProcessor {
     }
 
     fn do_swap_request(ka: &mut [KeyedAccount]) -> Result<(), InstructionError> {
-        if ka.len() < 7 {
+        if ka.len() < 6 {
             error!("Not enough accounts");
             Err(InstructionError::InvalidArgument)?
         }
-        Self::is_account_unallocated(&ka[1].account.data[..])?;
-        let mut to_trade = Self::deserialize_trade(&ka[2].account.data[..])?;
-        let mut from_trade = Self::deserialize_trade(&ka[3].account.data[..])?;
-        let mut to_trade_account = Self::deserialize_account(&ka[4].account.data[..])?;
-        let mut from_trade_account = Self::deserialize_account(&ka[5].account.data[..])?;
-        let mut profit_account = Self::deserialize_account(&ka[6].account.data[..])?;
+        Self::is_account_unallocated(&ka[0].account.data[..])?;
```

Comment:
What do you think about initializing some constants for all these `ka` keys?
Seems like it would be a little less fragile, and similar changes in the future would only have to happen in one place.

Reply:
@CriesofCarrots What did you have in mind?

Reply:
Something like system: https://github.com/solana-labs/solana/blob/2ab50cbae873754e9ed25bd15b589a8ccbcf71e4/runtime/src/system_program.rs#L9
It makes the code more verbose, but more meaningful than an int.
Your consts would obv have to be function-specific, since the keyed accounts are not consistent across instructions.